core/src/main/java/com/linecorp/armeria/server/Server.java

```diff
@@ -455,8 +455,21 @@ private void doStop(CompletableFuture<Void> future,
         }
 
         private void finishDoStop(CompletableFuture<Void> future) {
-            // TODO(trustin): Add shutdownBlockingTaskExecutorOnStop
-            // TODO(trustin): Count the pending blocking tasks and wait until it goes zero.
+            if (config.shutdownBlockingTaskExecutorOnStop()) {
+                if (config.blockingTaskExecutor() instanceof InterminableExecutorService) {
+                    final ExecutorService service =
+                            ((InterminableExecutorService) config.blockingTaskExecutor()).getExecutorService();
+                    try {
+                        service.shutdown();
+                        while (!service.isTerminated()) {
+                            service.awaitTermination(500, TimeUnit.MILLISECONDS);
```

Comment:
- You need to catch `InterruptedException` and swallow it because server shutdown should not be interrupted.
- Perhaps we can wait for more than 500ms? No need to wake up at all until termination. How about 1 day?